smart_card_connector_app/src/chrome-api-provider.js

```diff
@@ -0,0 +1,62 @@
+goog.provide('GoogleSmartCard.ChromeAPIProvider');
+
+goog.require('GoogleSmartCard.PcscLiteServerClientsManagement.ServerRequestHandler');
+
+goog.scope(function() {
+
+const GSC = GoogleSmartCard;
+
+/**
+ * @constructor
+ * @extends goog.Disposable
+ */
+GSC.ChromeAPIProvider = function(serverMessageChannel, serverReadinessTracker) {
+  this.serverRequester_ =
+      GSC.PcscLiteServerClientsManagement.ServerRequestHandler(
+          serverMessageChannel, serverReadinessTracker, 'chrome');
+
+  chrome.smartCardProviderPrivate.onEstablishContextRequested.addListener(
+      this.webapiEstablishContextListener_.bind(this));
+
+  // TODO: addOnDisposeCallback
+};
+
+goog.inherits(GSC.ChromeAPIProvider, goog.Disposable);
+
+/** @override */
+ChromeAPIProvider.prototype.disposeInternal = function() {
+  // TODO: add impl
+};
+
+
+/**
+ * @param {number} requestId
+ */
+ChromeAPIProvider.prototype.webapiEstablishContextListener_ = function(
+    requestId) {
+  const callArguments = [2, null, null];
+  const remoteCallMessage =
+      new GSC.RemoteCallMessage('SCardEstablishContext', callArguments);
+  // console.log("sending request", callArguments, serverRequestHandler);
+  const promise = this.serverRequester_.handleRequest(remoteCallMessage);
+  // console.log("webapi: SCardEstablishContext()");
+  promise.then(
```

Comment:
nit: I think you can use `await` to greatly simplify the code here. (You may wonder why a lot of existing code in this repo doesn't use it - but it's just because the project was mostly implemented before ES6 got adopted.)